src/controllers/auth.js

```diff
@@ -71,3 +81,57 @@ export const register = async (req, res) => {
     );
   }
 };
+
+export const forgetPassword = async (req, res) => {
+  const { email } = req.body;
+
+  const user = await findUserByEmail(email);
+
+  if (!user) formatResponse(res, StatusCodes.NOT_FOUND, null, "User not found");
+
+  const token = generateToken(user);
+
+  const info = await sendEmail(
+    email,
+    "Password reset",
+    `Click on the link to reset your password: http://localhost:3000/reset-password/${user.id}/${token}`
```

Comment:
It's better to not encode the base URL, so that we will be able to change it in the future after deployment. You can consider adding it in the env variables